tests/ludwig/utils/test_backward_compatibility.py

```diff
@@ -93,3 +110,82 @@ def test_audio_feature_backward_compatibility():
             "num_filter_bands": 80,
         },
     }
+
+
+def test_deprecated_field_aliases():
+    config = {
+        "ludwig_version": "0.4",
+        INPUT_FEATURES: [{"name": "num_in", "type": "numerical"}],
+        OUTPUT_FEATURES: [{"name": "num_out", "type": "numerical"}],
+        "training": {
+            "epochs": 2,
+            "eval_batch_size": 0,
+        },
+        HYPEROPT: {
+            "parameters": {
+                "training.learning_rate": {
+                    "space": "loguniform",
+                    "lower": 0.001,
+                    "upper": 0.1,
+                },
+            },
+            "goal": "minimize",
+            "sampler": {"type": "grid", "num_samples": 2, "scheduler": {"type": "fifo"}},
+            "executor": {
+                "type": "grid",
+                "search_alg": "bohb",
+            },
+        },
+    }
+
+    updated_config = upgrade_to_latest_version(config)
+
+    assert updated_config["input_features"][0][TYPE] == NUMBER
+    assert updated_config["output_features"][0][TYPE] == NUMBER
+
+    assert "training" not in updated_config
+    assert updated_config[TRAINER]["epochs"] == 2
+    assert updated_config[TRAINER][EVAL_BATCH_SIZE] is None
+
+    hparams = updated_config[HYPEROPT]["parameters"]
+    assert "training.learning_rate" not in hparams
+    assert "trainer.learning_rate" in hparams
+
+    assert "sampler" not in updated_config[HYPEROPT]
+
+    assert updated_config[HYPEROPT]["executor"]["type"] == "ray"
+    assert "num_samples" in updated_config[HYPEROPT]["executor"]
+    assert "scheduler" in updated_config[HYPEROPT]["executor"]
```

Comment:
Instead of these explicit asserts, could we simply assert on the full config, or at least in addition? I find these asserts to be less readable for groc-ing the overall config change.

Reply:
I agree, but I just copied these tests over from test_defaults.py.  I didn't want to make this refactoring and rewrite the tests in the same PR, so I kept them as is.